usr/src/cmd/zfs/zfs_main.c

```diff
@@ -2330,6 +2331,9 @@ us_compare(const void *larg, const void *rarg, void *unused)
 			if (rv64 != lv64)
 				rc = (rv64 < lv64) ? 1 : -1;
 			break;
+
+		default:
+			break;
```

Comment:
Looks like you can sort by a different (invalid) column, and it does ignore it.  I guess that's a different bug.  Seems like it should be validating the sort columns against us_field_names[] like it does the output (-o) columns.